torch/utils/_pytree.py

```diff
@@ -510,7 +511,7 @@ def __repr__(self, indent: int = 0) -> str:
         repr_suffix: str = f"{children_specs_str}])"
         return repr_prefix + repr_suffix
 
-    def is_leaf(self) -> bool:
+    def is_leaf(self) -> TypeGuard["LeafSpec"]:
```

Comment:
Yeah, so narrow self is explicitly not supported in the TypeGuard PEP. How annoying... Feel free to revert this change then.